azure-iot-device/tests/iothub/pipeline/test_pipeline_stages_iothub_mqtt.py

```diff
@@ -394,13 +450,99 @@ def test_operation_succeeds(self, params, stage, stages_configured_for_both, op)
     },
+    {
+        "name": "send telemetry with multiple system properties",
+        "stage_type": "device",
+        "op_class": pipeline_ops_iothub.SendD2CMessageOperation,
+        "op_init_kwargs": {
+            "message": Message(
+                fake_message_body, message_id=fake_message_id, content_type=fake_content_type
+            )
+        },
+        "topic": "devices/{}/messages/events/{}&{}".format(
+            fake_device_id, fake_message_id_encoded, fake_content_type_encoded
+        ),
+        "publish_payload": fake_message_body,
+    },
+    {
+        "name": "send telemetry with only single user property",
+        "stage_type": "device",
+        "op_class": pipeline_ops_iothub.SendD2CMessageOperation,
+        "op_init_kwargs": {
+            "message": create_message_with_user_properties(fake_message_body, is_multiple=False)
+        },
+        "topic": "devices/{}/messages/events/{}".format(
+            fake_device_id, fake_message_user_property_1_encoded
+        ),
+        "publish_payload": fake_message_body,
+    },
+    {
+        "name": "send telemetry with only multiple user properties",
+        "stage_type": "device",
+        "op_class": pipeline_ops_iothub.SendD2CMessageOperation,
+        "op_init_kwargs": {
+            "message": create_message_with_user_properties(fake_message_body, is_multiple=True)
+        },
+        # For more than 1 user property the order could be different, creating 2 different topics
+        "topic1": "devices/{}/messages/events/{}&{}".format(
+            fake_device_id,
+            fake_message_user_property_1_encoded,
+            fake_message_user_property_2_encoded,
+        ),
+        "topic2": "devices/{}/messages/events/{}&{}".format(
+            fake_device_id,
+            fake_message_user_property_2_encoded,
+            fake_message_user_property_1_encoded,
+        ),
+        "publish_payload": fake_message_body,
+    },
+    {
+        "name": "send telemetry with 1 system and 1 user property",
+        "stage_type": "device",
+        "op_class": pipeline_ops_iothub.SendD2CMessageOperation,
+        "op_init_kwargs": {
+            "message": create_message_with_system_and_user_properties(
+                fake_message_body, is_multiple=False
+            )
+        },
+        "topic": "devices/{}/messages/events/{}&{}".format(
+            fake_device_id, fake_message_id_encoded, fake_message_user_property_1_encoded
+        ),
+        "publish_payload": fake_message_body,
+    },
+    {
+        "name": "send telemetry with multiple system and multiple user properties",
+        "stage_type": "device",
+        "op_class": pipeline_ops_iothub.SendD2CMessageOperation,
+        "op_init_kwargs": {
+            "message": create_message_with_system_and_user_properties(
+                fake_message_body, is_multiple=True
+            )
+        },
+        # For more than 1 user property the order could be different, creating 2 different topics
```

Comment:
>For more than 1 user property the order could be different, [](start = 10, length = 59)

You should document which version of Python puts these out-of-order.  IIRC, it's 3.4, right?  That will be EOL soon enough.

Reply:
actually for me it happened for 2.7 , 3.4 and 3.5 , so better not to mention explicitly.

---
In reply to: [309384147](https://github.com/Azure/azure-iot-sdk-python-preview/pull/153#discussion_r309384147) [](ancestors = 309384147)